src/parser/shared/normalizers/Channeling.ts

```diff
@@ -279,6 +304,73 @@ export function buffChannelSpec(spellId: number): ChannelSpec {
   };
 }
 
+/**
+ * Helper to create a channel spec handler for Evokers empowered spells.
+ *
+ * This handler works by handling EmpowerStart events with the given guid, and then scanning forward for the
+ * matched EmpowerEnd, and then making the pair of beginchannel and endchannel events based on them.
+ * Sometimes Empowers don't produce EmpowerEnd, mainly when cancelling the spell. To account for this
+ * we will also cancel the channel on next cast, begincast or if buff is removed(fire breath).
+ *
+ * Tipped Empowers don't produce a EmpowerStart event so we use this event to filter those out.
+ *
+ * @param spellId the guid for the tracked Cast and EmpowerStart/EmpowerEnd events.
+ */
+export function empowerChannelSpec(spellId: number): ChannelSpec {
+  const guids = [spellId];
+  const handler: ChannelHandler = (
+    event: AnyEvent,
+    events: AnyEvent[],
+    eventIndex: number,
+    state: ChannelState,
+  ) => {
+    if (
+      event.type === EventType.EmpowerStart ||
+      event.type === EventType.Cast ||
+      event.type === EventType.RemoveBuff
+    ) {
+      // do standard start channel stuff
+      cancelCurrentChannel(event, state);
+      if (event.type === EventType.EmpowerStart) {
+        beginEmpowerCurrentChannel(event, state);
+      }
+      // now scan ahead for the matched empowerend, cast or begincast and end the channel at it
+      for (let idx = eventIndex + 1; idx < events.length; idx += 1) {
+        const laterEvent = events[idx];
+        if (
+          HasAbility(laterEvent) &&
+          laterEvent.ability.guid === spellId &&
+          laterEvent.type === EventType.EmpowerEnd
+        ) {
+          endCurrentChannel(laterEvent, state);
+          break;
+        } else if (
+          laterEvent.type === EventType.BeginCast ||
+          (laterEvent.type === EventType.Cast &&
+            isRealCast(laterEvent) &&
+            laterEvent.timestamp > event.timestamp &&
+            event.sourceID === playerInfo.combatant.id)
```

Comment:
I don't think this is actually what you're after, this _should_ return `undefined` due to the implementation.

Reply:
You were correct, changed to `playerInfo.playerId` - now properly returns playerID.